### PR TITLE
Free trial

### PR DESCRIPTION
**Pull request by averyschreiner**

### CODE REVIEW

File: Controllers/IndexController.cs [hunk 1]
```diff
@@ -42,8 +44,8 @@ public IActionResult ConfirmationEmail()
     [Route("/createprofile")]
     public IActionResult CreateProfile()
     {
-        var id = User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.NameIdentifier)?.Value!;
-        var family = _familyService.GetById(id).Data;
+        var email = User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Email)?.Value;
```
Comment (Pchristy03) [inline Controllers/IndexController.cs:112]: Is it safe to say this wont be null? Maybe we make we should check if its null before assigning it below?

File: ViewModels/AddressViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+public class AddressViewModel
+{
+    public string? StreetAddress { get; set; }
+    public string? AptOrUnitNumber { get; set; }
+    public string? CityOrTown { get; set; }
+    public string? StateProvinceRegion { get; set; }
+    public string? PostalCode { get; set; }
+    public string? Country { get; set; }
+}
```
Comment (Pchristy03) [inline ViewModels/AddressViewModel.cs]: Should these match the nullability of the original model?

File: Controllers/IndexController.cs [hunk 3]
```diff
@@ -270,11 +272,16 @@ int GetMealTypeOrder(string mealType)
             };
         }
 
+        var firstJobRun = _userStatusService.CalculateFirstJobRun(family.CreatedAt, family.GenerationDay, family.GenerationTime);
```
Comment (Pchristy03) [inline Controllers/IndexController.cs]: We should probably null check here just in case

File: Models/JobRecordModel.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+public class JobRecordModel
+{
+    public required string JobId { get; set; }
+    public required string FamilyId { get; set; }
+    public required DateTime StartTime { get; set; }
+    public required DateTime EndTime { get; set; }
+    public required TimeSpan Duration { get; set; }
+    public required JobStatus JobStatus { get; set; }
+    public required JobType JobType { get; set; }
+}
```
Comment (Pchristy03) [inline Models/JobRecordModel.cs:16]: Just a reminder to add these into the prod DB before we merge this in

**Comment by Pchristy03** [comment]
There is a lot of good stuff in here! Excited to see it all in action, nice work!

File: Controllers/IndexController.cs (hunk 1 above)
Comment (averyschreiner) [reply Controllers/IndexController.cs:112]: _should_ never be null as they will need to sign up with an email wether they use google auth or normal auth0 sign up

File: ViewModels/AddressViewModel.cs (hunk 2 above)
Comment (averyschreiner) [reply ViewModels/AddressViewModel.cs]: ideally yes, but we do all needed null checking in the controller. The form submission was being difficult if this model was not nulled out and if they chose not to provide their address

File: Controllers/IndexController.cs (hunk 3 above)
Comment (averyschreiner) [reply Controllers/IndexController.cs]: if they aren't signed up, when clicking on the overview page they will be redirected to create profile page

if they are signed up, they will always have first job run

File: Models/JobRecordModel.cs (hunk 4 above)
Comment (averyschreiner) [reply Models/JobRecordModel.cs:16]: noted!

**Pull request merged**